File: aula03 16_09/Java/Animal.java
public class Animal {
    protected String especie;
    protected String nome;

    public Animal(String especie, String nome) {
        this.especie = especie;
        this.nome = nome;
    }

    public String emitirSom(){
        return ""; //Método a ser sobreescrito
    }
}

class Cachorro extends Animal{

    public Cachorro(String nome) {
        super("Cachorro", nome);
    }

    @Override
    public String emitirSom() {
        return "Au Au";
    }
}

class Gato extends Animal{

    public Gato(String nome){
        super("Gato", nome);
    }

    @Override
    public String emitirSom() {
        return "Miau";
    }
}

File: aula03 16_09/Python/Animal.py
class Animal:
    def __init__(self, especie, nome):
        self.especie = especie
        self.nome = nome
    def emitir_som(self):
        pass #Método a ser implementado nas subclasses

class Cachorro(Animal):
    def emitir_som(self):
        return 'Au au'

class Gato(Animal):
    def emitir_som(self):
        return 'Miau'

if __name__ == '__main__':

    cachorro = Cachorro('Cachorro', 'Rex')
    gato = Gato('Gato', 'Michonne')
    print(cachorro.emitir_som())
    print(gato.emitir_som())
